calico/openstack/t_etcd.py

```diff
@@ -67,19 +69,25 @@ def initialize(self):
         self.client = etcd.Client(host=cfg.CONF.calico.etcd_host,
                                   port=cfg.CONF.calico.etcd_port)
 
-        # Spawn a green thread for periodically resynchronizing etcd against
-        # the OpenStack database.
-        eventlet.spawn(self.periodic_resync_thread)
-
         # We will remember the OpenStack security group data, between resyncs,
         # so that we can generate profiles when needed for new or updated
         # endpoints.  Start off with an empty set.
         self.sgs = {}
+        # Semaphore to protect all access to the sgs dict.
+        self._sgs_semaphore = Semaphore(1)
```

Comment:
Ah screw it, while it's not necessary it's not like it hurts more than the alternative. Let's be defensive and worry about it later.